doc/rpc_marshalable_objects.md

```diff
@@ -7,12 +7,38 @@ StreamJsonRpc allows transmitting marshalable objects (i.e., objects implementin
 
 Marshalable interfaces must:
 
-1. Extend `IDisposable`.
+1. Extend `IDisposable` (unless interface is call-scoped).
 1. Not include any properties.
 1. Not include any events.
 
 The object that implements a marshalable interface may include properties and events as well as other additional members but only the methods defined by the marshalable interface will be available on the proxy, and the data will not be serialized.
 
+## Call-scoped vs. explicitly scoped
+
+### Explicit lifetime
+
+An RPC marshalable interface has an explicit lifetime by default.
+This means that the receiver of a marshaled object owns its lifetime, which may extend beyond an individual RPC call.
+Memory for the marshaled object and its proxy are not released until the receiver either disposes of its proxy or the JSON-RPC connection is closed.
```

Comment:
It would be good to mention that JsonRpc connection gets closed if original service requested is disposed since that could be a common pitfall.

Reply:
The concept of *services* isn't in JSON-RPC at all. That's a brokered service concept which is defined in the vs-servicehub repo. Most proxy-generating APIs in JsonRpc itself do _not_ close the connection. But the `static JsonRpc.Attach<T>` method _does_ return a proxy whose disposal closes the connection, and that is documented on that particular API.